modules/um_callerid.c

```diff
@@ -227,7 +239,54 @@ h_hdl_privmsg_user(void *vdata)
 	data->approved = ERR_TARGUMODEG;
 }
 
+static void
+check_umode_change(void *vdata)
+{
+	hook_data_umode_changed *data = (hook_data_umode_changed *)vdata;
+	bool changed = false;
+	struct Client *source_p = data->client;
+
+	if (!MyClient(source_p))
+		return;
+
+	if (data->oldumodes & UMODE_OPER && !IsOper(source_p))
+		source_p->umodes &= ~user_modes['M'];
+
+	changed = ((data->oldumodes ^ source_p->umodes) & user_modes['M']);
+
+	if (changed && source_p->umodes & user_modes['M'])
+	{
+		if (!HasPrivilege(source_p, "oper:message"))
+		{
+			sendto_one_notice(source_p, ":*** You need oper:message privilege for +M");
```

Comment:
could we also have this mode timeout like override does?

Reply:
I'd rather timeout behaviors be implemented client side if desired. Timeouts at best lead to ugly hacks in bots